Source/Core/Core/AchievementManager.cpp

```diff
@@ -295,12 +318,23 @@ void AchievementManager::DoFrame()
   if (!IsGameLoaded() || !Core::IsCPUThread())
     return;
   {
+#ifdef RC_CLIENT_SUPPORTS_RAINTEGRATION
+    {
+      std::lock_guard lg{m_memory_lock};
+      Core::System* system = m_system.load(std::memory_order_acquire);
+      if (!system)
+        return;
+      Core::CPUThreadGuard thread_guard(*system);
+      u32 ram_size = system->GetMemory().GetRamSizeReal();
+      if (m_cloned_memory.size() != ram_size)
+        m_cloned_memory.resize(ram_size);
+      Core::System::GetInstance().GetMemory().CopyFromEmu(m_cloned_memory.data(), 0x80000000U,
```

Comment:
You can use `system` here. Please also note that there's still an unresolved comment regarding this line from my review back in January.